plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src-test/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineXsltTransformTests.java

```diff
@@ -0,0 +1,165 @@
+    assertEquals("java8", getRuntimeElement(transformed));
+  }
+
+  @Test
+  public void testRemoveBare() throws IOException, TransformerException {
+    Document transformed = transform("/xslt/removeJava8Runtime.xsl",
+        "<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\"/>");
+    assertNull(getRuntimeElement(transformed));
+  }
+
+  @Test
+  public void testRemovePreservesNonJava8Runtime() throws IOException, TransformerException {
+    Document transformed = transform("/xslt/removeJava8Runtime.xsl",
+        "<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime>XXX</runtime></appengine-web-app>");
+    assertEquals("XXX", getRuntimeElement(transformed));
+  }
+
+  @Test
+  public void testRemovePreservesOrder() throws IOException, TransformerException {
+    Document transformed = transform("/xslt/removeJava8Runtime.xsl",
+        "<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<threadsafe>true</threadsafe>" + "<runtime>java8</runtime>"
+            + "<sessions-enabled>true</sessions-enabled>" + "</appengine-web-app>");
+    Element documentElement = transformed.getDocumentElement();
+    NodeList childElements = documentElement.getChildNodes();
+    assertEquals("appengine-web-app", documentElement.getTagName());
+    assertEquals(2, childElements.getLength());
+    assertEquals("threadsafe", ((Element) childElements.item(0)).getTagName());
+    assertEquals("true", childElements.item(0).getTextContent());
+    assertEquals("sessions-enabled", ((Element) childElements.item(1)).getTagName());
+    assertEquals("true", childElements.item(1).getTextContent());
+  }
+
+  @Test
+  public void testRemoveExistingJava8Runtime() throws IOException, TransformerException {
+    Document transformed = transform("/xslt/removeJava8Runtime.xsl",
+        "<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">"
+            + "<runtime>java8</runtime></appengine-web-app>");
+    assertNull(getRuntimeElement(transformed));
+  }
+
+  private Document transform(String templateFile, String inputValue) {
+    try {
+      URL xslPath = AppEngineXsltTransformTests.class.getResource(templateFile);
+      if (xslPath == null) {
+        File xslFile = new File("." + templateFile);
+        assertTrue(xslFile.exists());
+        xslPath = xslFile.toURI().toURL();
+      }
+      InputStream documentStream =
+          new ByteArrayInputStream(inputValue.getBytes(StandardCharsets.UTF_8));
+      InputStream transformed = Xslt.applyXslt(documentStream, xslPath.openStream());
+      DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+      builderFactory.setNamespaceAware(true);
+      DocumentBuilder builder = builderFactory.newDocumentBuilder();
+      return builder.parse(transformed);
+    } catch (IOException | TransformerException | ParserConfigurationException | SAXException ex) {
```

Comment:
Are you ok `throws Exception`?  Otherwise means littering the method signatures with `throws W, X, Y, Z` which seemed like noise.